pkg/web/oauthclient/oauthclient.go

```diff
@@ -51,6 +52,16 @@ func (oc *OAuthClient) getMountPath() string {
 	return path
 }
 
+func (oc *OAuthClient) attachTLSClient(c echo.Context) context.Context {
+	transport := &http.Transport{
+		TLSClientConfig: oc.config.TLSClientConfig,
+	}
```

Comment:
I would personally do this here;

```go
func (oc *OAuthClient) withTLSClientConfig(ctx context.Context) (context.Context, error) {
  config, err := oc.component.GetTLSClientConfig(ctx)
  if err != nil {
    return nil, err
  }
  return context.WithValue(ctx, oauth2.HTTPClient, &http.Client{
    Transport: &http.Transport{
      TLSClientConfig: config,
    }
  }), nil
}
```

and use as;

```go
ctx, err := oc.withTLSClientConfig(c.Request().Context())
if err != nil {
  ...
}
..., err := oc.oauth(c).TokenSource(ctx, token).Token()
```

Because;
- this decouples echo context from configuration
- it uses the request context for getting TLS client config, which may become useful in the future
- it feels more idiomatic to derive contexts with the with-pattern

Reply:
Oh indeed! I completely overlooked that the `OAuthClient` had a reference to the component.